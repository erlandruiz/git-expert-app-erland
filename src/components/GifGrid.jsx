

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

import PropTypes from 'prop-types';





export const GifGrid = ({category}) => {

  const {images, isLoading} = useFetchGifs(category);
  console.log({images, isLoading})



    

  return (
    <>
    <h3>{category}</h3>

    {
      isLoading && (<h2>Cargando..</h2>)
     
    }
    




    <div className="card-grid">
      {
        // images.map((image)=>(
        //  <GifItem 
        //  key={image.id}
        //  title= {image.title}
        //  url={image.url}/>
        // )

        images.map((image)=>(
          <GifItem 
            key={image.id}
            {...image}
          
          />
        ))
        
        
      }
   
     
      
    </div>
   
    {/* <button onClick={()=> setCounter(counter+1)}>+1</button> */}


    {/* <p>Hola Erland</p> */}
    {/* {

    arrayNumeros.map(
        (numero) => (
        <p key = {numero}>{numero} </p>))
    
    } */}
    </>
  )
}

GifGrid.propTypes ={
  category: PropTypes.string.isRequired
}
