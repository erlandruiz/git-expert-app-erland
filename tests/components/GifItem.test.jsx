import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"



describe(' Pruebas en <GifItem />', () => { 

       const title = 'Erland';
        const url = 'https://www.erland.com/erland.jpg';


    test('debe de hacer match con el  snapshot', () => {  

     
          
       const {container} = render(<GifItem title={title} url = {url}/>);

       expect(container).toMatchSnapshot();

        
    });

    test('debe de mostrar la imagen con el URL y ALT indicado', () => { 
        render( <GifItem title={title} url={url}/> );
        // screen.debug();
        // console.log(screen.getByRole('img').src);
        // console.log(screen.getByRole('img').alt);
        // console.log(screen.getByRole('img'))

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
        
     });

     test('debe de mostrar el tÃ­tulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);

        expect(screen.getByText(title)).toBeTruthy();
      });
})