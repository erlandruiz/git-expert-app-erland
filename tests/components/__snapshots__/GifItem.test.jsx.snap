// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en <GifItem /> debe de hacer match con el  snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Erland"
      src="https://www.erland.com/erland.jpg"
    />
    <p>
      Erland
    </p>
  </div>
</div>
`;
